build/maven.go

```diff
@@ -174,43 +175,76 @@ func (mm *MavenModule) loadMavenHome() (mavenHome string, err error) {
 		if err != nil {
-			return "", err
+			return maven, err
 		}
-		output := strings.Split(strings.TrimSpace(stdout.String()), "\n")
-		// Finding the relevant "Maven home" line in command response.
-		for _, line := range output {
-			if strings.HasPrefix(line, "Maven home:") {
-				mavenHome = strings.Split(line, " ")[2]
-				if runtime.GOOS == "windows" {
-					mavenHome = strings.TrimSuffix(mavenHome, "\r")
-				}
-				mavenHome, err = filepath.Abs(mavenHome)
+		if !mm.extractorDetails.useWrapper {
+			mvnPath, err := exec.LookPath("mvn")
+			if err != nil || mvnPath == "" {
 				if err != nil {
-					return "", err
+					return "", errors.New(err.Error() + "\nHint: The mvn command may not be included in the PATH. Either add it to the path or set the M2_HOME environment variable value to the maven installation directory, which is the directory that includes the bin and lib directories.")
+				} else {
+					return "", errors.New("hint: The mvn command may not be included in the PATH. Either add it to the path or set the M2_HOME environment variable value to the maven installation directory, which is the directory that includes the bin and lib directories")
 				}
-				break
 			}
 		}
-		if mavenHome == "" {
-			return "", errors.New("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + stdout.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		versionOutput, err := mm.execMavenVersion(maven)
+		if err != nil {
+			return "", errors.New(err.Error() + "Could not find the location of the maven home directory, by running 'mvn --version' command. The command versionOutput is:\n" + versionOutput.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		}
+		// Finding the relevant "Maven home" line in command response.
+		mavenHome, err = mm.extractMavenPath(versionOutput)
+		if mavenHome == "" || err != nil {
+			if err != nil {
+				return "", errors.New(err.Error() + "Could not find the location of the maven home directory, by running 'mvn --version' command. The command versionOutput is:\n" + versionOutput.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+			} else {
+				return "", errors.New("Could not find the location of the maven home directory, by running 'mvn --version' command. The command versionOutput is:\n" + versionOutput.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+			}
```

Comment:
The code in this function includes a lot of duplications can be significantly simplified.